src/pymmcore_widgets/useq_widgets/_mda_sequence.py

```diff
@@ -509,54 +506,66 @@ def _settings_extensions(self) -> str:
 
     def _enable_af(self, state: bool, tooltip1: str, tooltip2: str) -> None:
         """Enable or disable autofocus settings."""
+        # enable autofocus axis widget
         self.af_axis.setEnabled(state)
         self.af_axis.setToolTip(tooltip1)
+        # enable autofocus per position checkbox
         self.stage_positions.af_per_position.setEnabled(state)
         self.stage_positions.af_per_position.setToolTip(tooltip2)
+        # hide the autofocus columns if autofocus per position is disabled
         if not state:
-            self.stage_positions.af_per_position.setChecked(state)
-        else:
-            # re-enable autofocus per position only if it was checked before
-            self.stage_positions.af_per_position.setChecked(self._use_af_per_pos)
+            # not simply calling self.stage_positions.af_per_position.setChecked(False)
+            # because we want to keep the previous state of the checkbox
+            self.stage_positions._on_af_per_position_toggled(False)
```

Comment:
this line smells a little funny.  we're using a private method of the stage_positions tab that looks mostly like an internal callback (that I wouldn't expect this other class to be using).  And it looks like we're just doing it to get that column hidden?  

though it might make for more code, one idea to keep these concerns more cleanly separated would be to add an eventFilter to the `PositionTable.af_per_position` Checkbox that listens for `QEvent.EnabledChange` events and hides the af column accordingly

Reply:
@tlambert03 I moved the logic in the `eventFilter` in the `CoreConnectedPositionTable`.